include/vinecopulib/vinecop/implementation/class.ipp

```diff
@@ -771,7 +874,12 @@ Vinecop::simulate(const size_t n,
                   const std::vector<int>& seeds) const
 {
   auto u = tools_stats::simulate_uniform(n, d_, qrng, seeds);
-  return inverse_rosenblatt(u, num_threads);
+  // inverse_rosenblatt() only works for continous models
+  auto actual_types = var_types_;
+  set_continuous_var_types();
```

Comment:
Can you explain why setting to continuous before sampling is the right thing to do?

Reply:
(I mean, I get why the inverse rosenblatt only exists for continuous models... but are we actually sampling what we thing that we are sampling when using this method for discrete variables?)

Reply:
Yes, the copula is always a continuous distribution. We just need to evaluate it differently to compute likelihoods and such. The sample should always be continuous; discreteness comes in when we transform with discrete margins.

Reply:
Right, thanks!